### PR TITLE
选项-显示新增显示扩展武将图片，设为默认开启

### PR DESCRIPTION
**Pull request by lieren2023**
选项-显示新增显示扩展武将图片，设为默认开启
#161

### CODE REVIEW

File: game/game.js [hunk 1]
```diff
@@ -7099,6 +7105,12 @@
 							else if(lib.character[name]){
 								nameinfo=lib.character[name];
 							}
+							else if(lib.config.show_extensionimage){
+								if(Object.keys(lib.characterPack).some(pack => Object.keys(lib.characterPack[pack]).contains(name))){
```
Comment (xiaoas) [inline game/game.js:7109]: `name in lib.characterPack[pack]`

File: game/game.js [hunk 2]
```diff
@@ -7099,6 +7105,12 @@
...
+									var pack=Object.keys(lib.characterPack).find(pack => Object.keys(lib.characterPack[pack]).contains(name));
```
Comment (xiaoas) [inline game/game.js:7110]: same as above

File: game/game.js (hunk 1 above)
Comment (xiaoas) [reply game/game.js:7109]: also this is a redundant `if` given the `var pack` statement below

File: game/game.js [hunk 3]
```diff
@@ -7099,6 +7105,12 @@
...
+									nameinfo=lib.characterPack[pack][name];
+								}
+							}
 							else if(name.indexOf('::')!=-1){
```
Comment (xiaoas) [inline game/game.js:7114]: the following name resolution fallbacks won't be reached if `lib.config.show_extensionimage` is set, which is incorrect behavior & render the config useless

**Review by xiaoas — CHANGES_REQUESTED** [review]

**Comment by Spmario233** [comment]
虽然不知道有啥意义 不过还是合了算了

**Pull request merged**